### PR TITLE
Support for Generic entities

### PR DESCRIPTION
**Pull request by athieriot**
Update most of the entityManager methods to support receiving an extra
ParameterizedType in case the user need to persist a Generic entity.

The type lost by erasure will then be reified using the informations
from the Parameterized type and the fields will be introspected as
normal.

Previous behaviour is not impacted and no breaking change was needed.

### CODE REVIEW

**Review by sinsir — APPROVED** [review]
Should any new classes contain your name as the @author tag?

Also, this was a massive review in itself, not helped by the misleading and irrelevant javadoc formatting issues - need to sort this out for future reviews?

**Comment by athieriot** [comment]
Yeah it would be nice to have the style fixed to the IntelliJ one but that would diverge a lot from the original repo

File: src/test/java/com/jmethods/catatumbo/entities/GenericParameterizedType.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.jmethods.catatumbo.entities;
```
Comment (sinsir) [inline src/test/java/com/jmethods/catatumbo/entities/GenericParameterizedType.java:16]: Should this have the copyright/licensing header ?

File: src/test/java/com/jmethods/catatumbo/entities/GenericParameterizedType.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.List;
+
+public class GenericParameterizedType implements ParameterizedType {
```
Comment (sinsir) [inline src/test/java/com/jmethods/catatumbo/entities/GenericParameterizedType.java]: @Author javadoc tag ?

**Review by edwardmlyte — APPROVED** [review]

**Pull request merged**